pkg/storage/unified/search/bleve.go

```diff
@@ -96,6 +100,37 @@ func (b *bleveBackend) GetIndex(ctx context.Context, key resource.NamespacedReso
 	return nil, nil
 }
 
+// updateIndexSizeMetric sets the total size of all file-based indices metric.
+func updateIndexSizeMetric(indexPath string, indexMetrics *resource.BleveIndexMetrics) {
+	if indexMetrics == nil {
+		return
+	}
+
+	for {
+		var totalSize int64
+
+		err := filepath.WalkDir(indexPath, func(path string, info os.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				fileInfo, err := info.Info()
+				if err != nil {
+					return err
+				}
+				totalSize += fileInfo.Size()
+			}
+			return nil
+		})
+
+		if err == nil {
```

Comment:
We should log the error so that there's a trace of problem, and operators don't need to wonder why the metric is not updated.